### PR TITLE
commit test

### PR DESCRIPTION
**Pull request by nat526**
* response.txt for question n_1
* ILinkedList
* IntAllocationBase

### CODE REVIEW

File: ILinkedList.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            return head
+        }
+        
+		virtual void remove(Node* erase_node){
+            Node *node = head;
+            if (head == erase_node){
+                // if there is just one node
+                // reinitialize the head
+                head->next = NULL
```
Comment (avraham-634) [inline ILinkedList.cpp]: What will happen if `head->next` is not `NULL`?

File: ILinkedList.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                head = NULL;
+            }
+            else{
+                // then, loop over nodes,
+                while(node->next = erase_node){
+                    // link the before node to the next node
+                    node->next = erase_node->next;
+                    // reinitialize the node
+                    erase_node->next = NULL;
+                    erase_node = NULL;
```
Comment (avraham-634) [inline ILinkedList.cpp]: Is there a bug here?

File: ILinkedList.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+    void ~ILinkedList(){
+            if (head == NULL)
+                return
+            Node *node = head;
+            while (node->next){
+                // loop over nodes,
+                // reinitialize the curent node
+                next_node = node->next
```
Comment (avraham-634) [inline ILinkedList.cpp]: Will this compile?

File: ILinkedList.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                node->next = NULL
+                //step over to the next node
+                node = next_node
```
Comment (avraham-634) [inline ILinkedList.cpp]: Is there a bug here?

File: IntAllocatorBase.cpp [hunk 5]
```diff
@@ -0,0 +1,46 @@
+class IntAllocatorBase {
+	public:
+		virtual int* allocate(size_t size) = 0;
```
Comment (avraham-634) [inline IntAllocatorBase.cpp]: Where is the implementation for this pure virtual method?

File: IntAllocatorBase.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		
+		virtual void deallocate(int* ptr, size_t size) = 0;
```
Comment (avraham-634) [inline IntAllocatorBase.cpp]: Same here?

File: IntAllocatorBase.cpp [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	private:
+		std::array<int, 100> m_buffer;
+
+        void insert(int val){
+            // allocate new data memory with size +1
+            ptr = allocate(m_buffer.size+1);
```
Comment (avraham-634) [inline IntAllocatorBase.cpp]: Will this compile?

**Review by avraham-634 — COMMENTED** [review]
@nat526 - did you try to compile the code?

**Comment by nat526** [comment]
> @nat526 - did you try to compile the code?

I not compile it. I stayed on 'dry' run.
I can compile it and repair somethings, if you want

File: IntAllocatorBase.cpp (hunk 5 above)
Comment (nat526) [reply IntAllocatorBase.cpp]: I understood from the notice that i can use it as an implemented func...
I thought that i have to implement the add/remove func and use the de/allocat func

**Comment by avraham-634** [comment]
> > @nat526 - did you try to compile the code?
> 
> I not compile it. I stayed on 'dry' run. I can compile it and repair somethings, if you want

Please do.

**Pull request closed without merging**